RecoLocalTracker/Phase2TrackerRecHits/test/RecHitsValidation.cc

```diff
@@ -264,13 +487,37 @@ std::map< unsigned int, RecHitHistos >::iterator Phase2TrackerRecHitsValidation:
     return insertedIt.first;
 }
 
+
 unsigned int Phase2TrackerRecHitsValidation::getLayerNumber(const DetId& detid, const TrackerTopology* topo) {
     if (detid.det() == DetId::Tracker) {
-        if (detid.subdetId() == PixelSubdetector::PixelBarrel) return (topo->pxbLayer(detid));
-        else if (detid.subdetId() == PixelSubdetector::PixelEndcap) return (100 * topo->pxfSide(detid) + topo->pxfDisk(detid));
-        else return 999;
+        if (detid.subdetId() == PixelSubdetector::PixelBarrel) {
+          return (topo->pxbLayer(detid));
+        } else if (detid.subdetId() == PixelSubdetector::PixelEndcap) {
+          return (100 * topo->side(detid) + topo->pxfDisk(detid));
+        } else if (detid.subdetId() == StripSubdetector::TIB) {
+          return (topo->layer(detid));
+        } else if (detid.subdetId() == StripSubdetector::TOB) {
+          return (topo->layer(detid));
+        } else if (detid.subdetId() == StripSubdetector::TID) {
+          return (100 + topo->tidRing(detid));
+        } else if (detid.subdetId() == StripSubdetector::TEC) {
+          return (100 + topo->tecRing(detid));
+        } else {
+	  return 999;
+	}
```

Comment:
Can you Remind me why we need this TID, TOB TIB etc.. structure in this phase2 file ?